Assets/MixedRealityToolkit/Providers/BaseInputDeviceManager.cs

```diff
@@ -61,36 +155,97 @@ public abstract class BaseInputDeviceManager : BaseDataProvider<IMixedRealityInp
         /// <param name="useSpecificType">Only register pointers with a specific type.</param>
         protected virtual IMixedRealityPointer[] RequestPointers(SupportedControllerType controllerType, Handedness controllingHand)
```

Comment:
>RequestPointers [](start = 49, length = 15)

We haven't really done this (at all) for other parts, but since we're doing perf work in this case, consider adding something like:

[https://docs.unity3d.com/ScriptReference/Profiling.Profiler.BeginSample.html](https://docs.unity3d.com/ScriptReference/Profiling.Profiler.BeginSample.html)

So that we can measure the impact (in the Unity profiler) or having pointer caching vs not caching.